spec/cocoapods/generate/podfile_generator_spec.rb

```diff
@@ -57,15 +60,26 @@
     end
   end
 
-  describe_method 'podfiles_by_spec' do
-    let(:config_options) { { podspecs: [Pod::Spec.new(nil, 'A'), Pod::Spec.new(nil, 'B')] } }
-    before do
-      expect(podfile_generator).to receive(:podfile_for_spec).twice.and_wrap_original do |_original_method, spec, &_|
-        "Podfile for #{spec.name}"
+  describe_method 'podfiles_by_specs' do
+    context 'without single workspace' do
+      let(:config_options) { { podspecs: [Pod::Spec.new(nil, 'A'), Pod::Spec.new(nil, 'B')] } }
+      before do
+        expect(podfile_generator).to receive(:podfile_for_specs).twice.and_wrap_original do |_original_method, specs, &_|
+          "Podfile for #{specs.map(&:name).to_sentence}"
+        end
       end
+      it { should eq [Pod::Spec.new(nil, 'A')] => 'Podfile for A', [Pod::Spec.new(nil, 'B')] => 'Podfile for B' }
     end
 
-    it { should eq Pod::Spec.new(nil, 'A') => 'Podfile for A', Pod::Spec.new(nil, 'B') => 'Podfile for B' }
+    context 'with single workspace' do
+      let(:config_options) { { podspecs: [Pod::Spec.new(nil, 'A'), Pod::Spec.new(nil, 'B')], single_workspace: true } }
+      before do
+        expect(podfile_generator).to receive(:podfile_for_specs).and_wrap_original do |_original_method, specs, &_|
+          "Podfile for #{specs.map(&:name).to_sentence}"
+        end
+      end
+      it { should eq [Pod::Spec.new(nil, 'A'), Pod::Spec.new(nil, 'B')] => 'Podfile for A and B' }
```

Comment:
new test that verifies both specs are part of the same podfile for single workspaces.